### PR TITLE
Unit Edit

### PR DESCRIPTION
**Pull request by lawsj**
# Description

Added Default Min/Max values for edit units. 

Partially Addresses #1307

## Type of change

(Check the ones that apply by placing an "x" instead of the space in the [ ] so it becomes [x])

- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

(Note what you have done by placing an "x" instead of the space in the [ ] so it becomes [x]. It is hoped you do all of them.)

- [x] I have followed the [OED pull request](https://openenergydashboard.org/developer/pr/) ideas
- [x] I have removed text in ( ) from the issue request
- [x] You acknowledge that every person contributing to this work has signed the [OED Contributing License Agreement](https://openenergydashboard.org/developer/cla/) and each author is listed in the Description section.

## Limitations

Still need to add the 'default disable check' and the 'default date check.

### CODE REVIEW

**Comment by Rakesh-Ranga-Buram** [comment]
{C3EF6B16-27C6-4D6B-9054-2C69D17BDED8}
It would suggest that the minimum and maximum value input boxes be on the same row rather than a different row.

The improvements suggested by Steve to @mmehta2669 work, apply to this as well, like the default maximum and minimum should not be hardcoded, instead be the maximum and minimum value in javascript

Also, @mmehta2669 had similar changes on the branch unitMaxMinClient, I would request you to coordinate with Mark on this

**Comment by lawsj** [comment]
Changes made as requested from comment @Rakesh-Ranga-Buram. The changes are in the second commit.

File: src/client/app/types/redux/units.ts [hunk 1]
```diff
@@ -46,8 +46,10 @@ export interface UnitEditData {
 	displayable: DisplayableType;
 	preferredDisplay: boolean;
 	note: string;
-	defaultMeterMinimumValue:-999999999999,
-	defaultMeterMaximumValue:999999999999,
+	defaultMeterMinimumValue: number;
```
Comment (Rakesh-Ranga-Buram) [inline src/client/app/types/redux/units.ts:49]: This whole section UnitEditData can be commented out as it is not used elsewhere in the application

File: src/client/app/components/unit/EditUnitModalComponent.tsx [hunk 2]
```diff
@@ -450,6 +480,21 @@ export default function EditUnitModalComponent(props: EditUnitModalComponentProp
...
+							name='disableChecks'
+							type='select'
+							value={state.note}
+							onChange={e => handleStringChange(e)}>
+							<option value=''> Select an option </option>
+							<option value='No, only reject the bad reading(s)'> No, only reject the bad reading(s) </option>
+							<option value='No, reject all readings in batch'> No, reject all readings in batch </option>
+							<option value='Yes, do not check values'> Yes, do not check value </option>
+						</Input>
+					</FormGroup>
```
Comment (Rakesh-Ranga-Buram) [inline src/client/app/components/unit/EditUnitModalComponent.tsx]: For Disable checks, the expectation is to create an enum and options should loop through the enum and translate the enum keys to display values of the options

File: src/client/app/translations/data.ts [hunk 3]
```diff
@@ -128,20 +128,16 @@ const LocaleTranslationData = {
 		"default.file.size.limit": "Default File Size Limit",
 		"default.help.url": "Documentation URL",
 		"default.time.zone": "Default Time Zone",
-		"default.meter.minimum.value": "Default meter minimum reading value check",
-		"default.meter.maximum.value": "Default meter maximum reading value check",
-		"default.meter.minimum.date": "Default meter minimum reading date check",
-		"default.meter.maximum.date": "Default meter maximum reading date check",
```
Comment (Rakesh-Ranga-Buram) [inline src/client/app/translations/data.ts:134]: we don't need to edit date, just edit the maximum and minimum value like
"default.minimum.value": "Default minimum reading value check",
"default.maximum.value": "Default maximum reading value check"
and do the same for translations in different languages with a bulb icon.
Also when you do these changes, the previous instances of these keys must also be changed
And the lines where these translations are written should not change so please revert back the changes done

**Comment by Rakesh-Ranga-Buram** [comment]
The date checks are not required, so limit the changes to maximum value, minimum value and disable checks

**Comment by lawsj** [comment]
New set of comments address in commit 5. This includes removed unused instance of key values, removing everything related to 'reading date', and making the disable checks field an enum.

**Comment by Rakesh-Ranga-Buram** [comment]
All the code improvements are done on a different branch #30  to stay in sync with the team. Hence closing it

**Pull request closed without merging**